core/variant/array.cpp

```diff
@@ -47,6 +47,34 @@ class ArrayPrivate {
 	Vector<Variant> array;
 	Variant *read_only = nullptr; // If enabled, a pointer is used to a temporary value that is used to return read-only values.
 	ContainerTypeValidate typed;
+
+	/// structs
+
+	uint32_t struct_size = 0;
+	const StructMember *struct_members = nullptr;
```

Comment:
I know you are trying to simplify the code, but this should work like the proposal for performance reasons. With separate arrays, this means less cache lines loaded for the lookup, hence faster performance.